### ISSUE TITLE
Reconnect not working even when retry_first_connection=True

### ISSUE THREAD

**Issue by wiebeytec** [eclipse/paho.mqtt.python#264]
I was wondering, what is the idea behind [this except clause](https://github.com/eclipse/paho.mqtt.python/blob/e9914a759f9f5b8081d59fd65edfd18d229a399e/src/paho/mqtt/client.py#L896)?

I start paho with `loop_start()`, which implies [retry_first_connection=True](https://github.com/eclipse/paho.mqtt.python/blob/e9914a759f9f5b8081d59fd65edfd18d229a399e/src/paho/mqtt/client.py#L2649). However, if I block the server with iptables, using REJECT, I get a [raise here](https://github.com/eclipse/paho.mqtt.python/blob/e9914a759f9f5b8081d59fd65edfd18d229a399e/src/paho/mqtt/client.py#L895).

Why are only those three errors retried?

In other words: currently there seems to be a bug that `retry_first_connection` doesn't work.

**Comment by PierreF** [comment]
This only apply when connect_async is used. I assume that you get the error when calling connect(), which try to immediately connect.

About the except clause... I think it not useful currently. We should add the timeout option to create_connection for it to be useful.

**Comment by MattBrittan** [comment]
I'm going to close this due to inactivity (the code discussed has changed a lot since this issue was raised). If you are still experiencing the issue then please feel free to reopen. This is part of a general project to clean-up issues (which should make it simpler to identify priorities going forward).

**Comment by inad9300** [comment]
> This only apply when connect_async is used.

Worth mentioning in the docs.
